webClient/src/app/editor/code-editor/monaco/monaco.component.ts

```diff
@@ -192,13 +217,18 @@ export class MonacoComponent implements OnInit, OnChanges {
       // Method that will be executed when the action is triggered.
       // @param editor The editor instance is passed in as a convenience
       run: function (ed) {
-        let fileContext = self.editorControl.fetchActiveFile();
-        let sub = self.monacoService.saveFile(fileContext, self.editorControl.activeDirectory).subscribe(() => sub.unsubscribe());
+        self.saveFile();
         return null;
       }
     });
   }
 
+  saveFile() {
+    let fileContext = this.editorControl.fetchActiveFile();
+    let directory = fileContext.model.path || this.editorControl.activeDirectory;
+    let sub = this.monacoService.saveFile(fileContext, directory).subscribe(() => sub.unsubscribe());
```

Comment:
That trick with `subscribe/unsubscribe` is clumsy. We can fix `monacoService.saveFile()` in the future by adding `obs.complete()` after each call to `obs.next()`, so that `unsubscribe` won't needed anymore.

Reply:
> Good work. It works for me. See my comments above for issues I noticed.
> Also it would be nice if we could select files for compare, e.g. using right click menu as in VSCode:
> 
> 1. I can select a file using right click menu:
>    ![image](https://user-images.githubusercontent.com/34669449/132799382-a522a6af-b9e0-471a-93c2-01678fb473d6.png)
> 2. then I can right click on another file and compare it with selected
>    ![image](https://user-images.githubusercontent.com/34669449/132799460-bb8a5354-edec-4675-97ea-aeaffdb3170b.png)

Totally agree, would love it too! But I think I'll leave it for a different PR for times' sake

Reply:
I think `.complete` as a callback unsubscribes after` .next` so I think we accomplish the same thing here, except with this clumsy way, callers for `.next` don't have to remember to use `.complete` Let me know if it's more important performance-wise than I think it is

Also `.complete` doesn't get called in the error case (or at least not according to the docs?), so I'm not sure if that's also something that needs to be handled with `.error`

Reply:
Actually, this doesn't handle error either

Reply:
So because complete auto-unsubscribes, it would be something like

`let sub = self.saveFile(fileContext, directory).subscribe(() => {}, e => {}, () => {});`

if we wanted to be formal because we don't care about error and we don't need a result from next, all is handled upstream